Documentation/Installation/EnvironmentConfiguration.rst

```diff
@@ -2,35 +2,197 @@
+    :language: bash
+    :caption: <project-root>/.env
+
+The next step is to retrieve these values in the TYPO3 application bootstrap
+process. The best place for this is inside :file:`system/additional.php` (see
+:ref:`t3coreapi:configuration-files`). The PHP code for this could look like:
+
+..  literalinclude:: Environments/_additional.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Each environment would have its own :file:`.env` file, which is only stored on
+the corresponding target system. The development environment file could
+be saved as :file:`.env.example` or delivered as the default :file:`.env`
+in your project.
+
+..  todo: Should we have a distinct "Versioning" chapter?
+
+It is not recommended to store these :file:`.env` files in your version control
+system (e.g. Git). The main reason is that these files usually hold credentials
+or other sensitive information.
+
+You should only store environment-specific configuration values in such a
+configuration file. Do not use this to manage all the TYPO3 configuration options.
+Examples of well-suited configuration options:
+
+* :ref:`t3coreapi:password-policies`
+* :ref:`t3coreapi:error-handling-configuration`
+* :ref:`t3coreapi:typo3ConfVars_mail`
+* :ref:`t3coreapi:typo3ConfVars_sys_encryptionKey`
+* :ref:`t3coreapi:security-install-tool`
+* Settings, Tokens and URLs to additional services (Redis, Solr, ...)
+
+..  note::
+    Note that the URL of your environment must be configured through
+    :ref:`site configuration <t3coreapi:sitehandling>` variables, and
+    those can actually refer to environment variables as outlined in
+    :ref:`t3coreapi:sitehandling-using-env-vars`.
+
+The following sections describe this implementation process in depth.
+
+.. _environment-dotenv:
+
+.env / dotenv files
+===================
+
+A central advantage of :file:`.env` files is that environment variables can
+also be set in CLI context or injected via Continuous Integration/Deploment
+(CI/CD) systems (GitLab/GitHub) or even Webserver configuration. A central place
+for environment-specific configuration, which many other applications refer to
+(and is easy to locate) is also a best practice.
+
+To let your TYPO3 configuration parse keys and values stored of such a file,
+you need a library like https://github.com/symfony/dotenv/ or
+https://github.com/vlucas/phpdotenv/ and parse it in your :file:`system/additional.php`
+
+You can require these libraries through Packagist/Composer or by including the dependency
+in your project manually (legacy installation).
+
+Example for `symfony/dotenv`:
+
+..  literalinclude:: Environments/_dotenv-symfony.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Example for `vlucas/phpdotenv`:
+
+..  literalinclude:: Environments/_dotenv-vlucas.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Once this code has loaded the :php:`$_ENV` variables from the specified file,
```

Comment:
maybe better:

Once this code has loaded the content from .env file into $_ENV variables ...